R/ops-addition.r

```diff
@@ -56,7 +56,7 @@ add_period_to_date <- function(per, date) {
                 seconds = second(lt) + per@.Data)
 
   if (is.Date(date) && sum(new$sec, new$min, new$hour, na.rm = TRUE) != 0)
-    return(new)
+    return(as.POSIXct(new))
```

Comment:
Is it possible to fix this upstream? (i.e. in `update()`)?

Reply:
You mean let update.Date always return a POSIXct? Well, the update here is update.POSIXlt preceded by add_months. So also rewrite add_months to have Date output?

Reply:
Ok, if that's going to involve going down a rabbit hole, it's fine to fix here.